utils/cluster_sample.py

```
@@ -0,0 +1,29 @@
+from sklearn.mixture import GaussianMixture
+import numpy as np
+
+class RandCluster:
+    def __init__(self, n_components):
+        self.n_components = n_components
+
+    def rand_means(self,factor_means):
+        self.means = np.random.random(size = n_samples)
```

Ruff (0.8.2)

Undefined name `n_samples`

(F821)